### PR TITLE
update types for autosize

### PR DESCRIPTION
**Pull request by zikaari**
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://www.jacklmoore.com/autosize/
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

# Changes
- Add jQuery overloads
- Fixed return values of functions, they now return whatever type is passed in
- Module no longer pollutes global scope, consumers will now require `import autosize from 'autosize'`
- Added function comments (for better intellisense experience)

### CODE REVIEW

**Comment by NeekSandhu** [comment]
Forgot to mention, this PR is revision of #17879, which was closed because I deleted my fork and couldn't commit any changes.

File: types/autosize/tsconfig.json [hunk 1]
```diff
@@ -6,18 +6,19 @@
             "dom"
         ],
         "noImplicitAny": true,
+        "noEmit": true,
```
Comment (andy-ms) [inline types/autosize/tsconfig.json]: Don't modify this file (unless you are enabling `strictNullChecks`).

**Review by ghost — COMMENTED** [review]
If the library genuinely exposes itself as a global variable, we should respect that decision and keep the `export as namespace`. Otherwise people will be needlessly broken.

File: types/autosize/tsconfig.json (hunk 1 above)
Comment (NeekSandhu) [inline types/autosize/tsconfig.json]: Library does not only export a namespace, but a function and object with same identifier `autosize`.
Also the lib doesn't look global it appears to be UMD module.
https://github.com/jackmoore/autosize/blob/master/dist/autosize.js

Regarding tsconfig, I didn't know and I will fix that at once we settle the other concern

Comment (andy-ms) [inline types/autosize/tsconfig.json]: `export is namespace` is the correct pattern for UMD modules.

**Review by mhegazy — CHANGES_REQUESTED** [review]
Please address review comments in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18089#pullrequestreview-51811904

**Comment by NeekSandhu** [comment]
Everything looks green.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**